pkg/traceql/lexer.go

```diff
@@ -167,18 +171,34 @@ func (l *lexer) Lex(lval *yySymType) int {
 		return FLOAT
 	}
 
-	tokStrNext := l.TokenText() + string(l.Peek())
-	if tok, ok := tokens[tokStrNext]; ok {
-		l.Next()
-		l.parsingAttribute = startsAttribute(tok)
-		return tok
+	// look for combination tokens starting with 2 and working up til there is no match
```

Comment:
Interesting.  I think this works because `!<` is a valid prefix of `!<<` (for example)?  If we had a three character token that didn't overlap a two character token, not sure it would work.  Not an issue, just thinking out loud and maybe a comment to explain for next time.

Reply:
yes, this is to disambiguate common prefixes for 2 and 3 char tokens. it's a good point that it would not find 3 char tokens that did not have a 2 char prefix.